syft/execution/plan.py

```diff
@@ -201,9 +203,16 @@ def build(self, *args):
             for arg in args
         )
 
+        # Wrap arguments that require gradients with AutogradTensor
+        # Note that AutogradTensor comes before Placeholder in the chain
+        # so that all operations that happen inside AutogradTensor are recorded by Placeholder
```

Comment:
Not sure I follow this comment. What wouldn't work if the chain was `Placeholder > AutogradTensor > ... other tensors in chain ...`?

Reply:
If placeholder comes before autograd in chain, trace will record `backward` method call on autograd tensor, but not anything that happens inside backward. So trace will just have `backward` command, which won't work in torchscript.
We'll probably want to do same thing with other custom tensor types, so all ops are captured right before the torch tensor.

Reply:
Hmm, okay. Let's leave that for a later PR.